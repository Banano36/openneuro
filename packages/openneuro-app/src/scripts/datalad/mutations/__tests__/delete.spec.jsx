import React from 'react'
import { shallow } from 'enzyme'
import DeleteDataset from '../delete.jsx'
import DeleteDir, { deleteFilesReducer } from '../delete-dir.jsx'

describe('DeleteDataset mutation', () => {
  it('renders with common props', () => {
    const wrapper = shallow(<DeleteDataset datasetId="ds001" />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('DeleteDir mutation', () => {
  it('renders with common props', () => {
    const wrapper = shallow(
      <DeleteDir
        datasetId="ds002"
        fileTree={{
          files: [],
          directories: [],
          path: '',
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  describe('deleteFilesReducer', () => {
    it('removes all files in given fileTree from dataset.draft', () => {
      const fileTree = {
        files: [{ filename: 'A' }],
        directories: [
          {
            files: [{ filename: 'B.ext' }, { filename: 'C.ext' }],
            directories: [],
            path: 'root_dir:removed_dir:dir1',
          },
          {
            files: [{ filename: 'D' }],
            directories: [
              {
                files: [{ filename: 'E.ext' }, { filename: 'F.ext' }],
                directories: [],
                path: 'root_dir:removed_dir:dir2:dir3',
              },
            ],
            path: 'root_dir:removed_dir:dir2',
          },
        ],
        path: 'root_dir:removed_dir',
      }
      const draft = {
        files: [
          { filename: 'root_dir/removed_dir/A' },
          { filename: 'root_dir/removed_dir/dir1/B.ext' },
          { filename: 'root_dir/removed_dir/dir1/C.ext' },
          { filename: 'root_dir/removed_dir/dir2/D' },
          { filename: 'root_dir/removed_dir/dir2/dir3/E.ext' },
          { filename: 'root_dir/removed_dir/dir2/dir3/F.ext' },
          { filename: 'root_dir/last_lonely_file' },
        ],
      }
      const updatedDraft = deleteFilesReducer(fileTree, draft)
      expect(updatedDraft.files.length).toBe(1)
      expect(updatedDraft.files[0].filename).toBe('root_dir/last_lonely_file')
    })
  })
})
