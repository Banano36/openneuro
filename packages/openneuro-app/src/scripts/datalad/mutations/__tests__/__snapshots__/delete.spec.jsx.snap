// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDataset mutation renders with common props 1`] = `
<<<<<<< HEAD
<ContextConsumer>
  <Component />
</ContextConsumer>
=======
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "deleteDataset",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "id",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "deleteDataset",
                },
                "selectionSet": undefined,
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 69,
        "start": 0,
      },
    }
  }
>
  <Component />
</Mutation>
>>>>>>> b95ea011... update permissions to array with id
`;

exports[`DeleteDir mutation renders with common props 1`] = `
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "deleteFiles",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "datasetId",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "datasetId",
                      },
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "files",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "files",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "deleteFiles",
                },
                "selectionSet": undefined,
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "datasetId",
                },
              },
            },
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "FileTree",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "files",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 120,
        "start": 0,
      },
    }
  }
>
  <Component />
</Mutation>
`;
